src/main/java/com/github/charlyb01/music_control/client/MusicControlClient.java

```diff
@@ -157,18 +157,20 @@ public void onInitializeClient() {
 
             while (volumeUp.wasPressed()) {
                 float volume = client.options.getSoundVolume(SoundCategory.MUSIC);
-                volume = Math.min(volume + 0.01F, 1.0F);
+                float newVolume = Utils.roundUp(volume + (float) ModConfig.get().volumeIncrement / 100, 2);
+                volume = Utils.clamp(newVolume, 0.0F, ModConfig.get().allowHighVolume ? 2.0F : 1.0F);
                 client.options.setSoundVolume(SoundCategory.MUSIC, volume);
                 client.options.write();
-                Utils.print(client, new TranslatableText("music.volume", (int) (100 * volume)));
+                Utils.print(client, new TranslatableText("music.volume", (int) (100 * volume)).append("%"));
             }
 
             while (volumeDown.wasPressed()) {
                 float volume = client.options.getSoundVolume(SoundCategory.MUSIC);
-                volume = Math.max(volume - 0.01F, 0.0F);
+                float newVolume = Utils.roundDown(volume - (float) ModConfig.get().volumeIncrement / 100, 2);
+                volume = Utils.clamp(newVolume, 0.0F, ModConfig.get().allowHighVolume ? 2.0F : 1.0F);
                 client.options.setSoundVolume(SoundCategory.MUSIC, volume);
                 client.options.write();
-                Utils.print(client, new TranslatableText("music.volume", (int) (100 * volume)));
+                Utils.print(client, new TranslatableText("music.volume", (int) (100 * volume)).append("%"));
```

Comment:
You should put it in the lang file, not directly in code ;)

Reply:
I tried that, it breaks the string formating

Reply:
Yes, it's because the % is used to tell minecraft there will be a variable after. But you still can say you want just a % displayed with %%. By the way, you can also append it the previous one used for the variable so that there is no space in-between (like %s%%)

Reply:
Ok, I'll make the edits

Reply:
Fixed in https://github.com/sf-inc/music_control/pull/20/commits/e714d8f059d107571ddb28441d9fa7ae48d34613